### PR TITLE
Library.io auth

### PR DESCRIPTION
**Pull request by CollinAvidano**
Should make all of the changes necessary for Apiary to support the jedi changes for the autodesk to library.io authorization

### CODE REVIEW

File: resources/views/autodesk.blade.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    Autodesk Access Request
+    @endcomponent
+
+    <p>
+{{ $message }}
+</p>
+
+<a href="/">Go back to the dashboard</a>
+
+                               @endsection
```
Comment (JoshOldenburg) [inline resources/views/autodesk.blade.php:18]: Checks don't enforce style in views. The indentation is weird here. The content is fine though.

File: resources/js/components/UserEditForm.vue [hunk 2]
```diff
@@ -232,6 +254,8 @@ export default {
       ],
       clickUpEmailOptions: [],
       clickUpEmailInDatabase: null,
+      AutodeskEmailOptions: [],
+      AutodeskEmailInDatabase: null,
```
Comment (JoshOldenburg) [inline resources/js/components/UserEditForm.vue:258]: camelCase, please. Checks don't enforce style in Vue files.

File: resources/js/components/UserEditForm.vue [hunk 3]
```diff
@@ -273,6 +303,14 @@ export default {
...
             this.feedback = 'Saved!';
           }
+
+          if (this.user.autodesk_email !== this.AutodeskEmailInDatabase && this.user.autodesk_email && this.user.autodesk_email.length > 0) {
+            this.AutodeskEmailInDatabase = this.user.autodesk_email;
+            this.feedback = 'Saved! Look out for an email from Autodesk in the next few minutes.'
+          } else {
+            this.feedback = 'Saved!';
+          }
+
```
Comment (JoshOldenburg) [inline resources/js/components/UserEditForm.vue:313]: The feedback for the clickup field will always be overwritten by the autodesk feedback. For instance, if you set a clickup email and not an autodesk email, `this.feedback` will be set on line 302 and then overwritten by line 311. The feedback from both should be merged.

File: app/Nova/User.php [hunk 4]
```diff
@@ -196,6 +197,17 @@ public function fields(Request $request): array
...
+                    ->hideFromIndex()
+                    ->rules('nullable', 'max:255', 'email')
+                    ->creationRules('unique:users,autodesk_email')
+                    ->updateRules('unique:users,autodesk_email,{{resourceId}}'),
+
+                    Boolean::make('Autodesk Invite Pending', 'autodesk_invite_pending')
+                    ->hideFromIndex()
+                    ->help('This flag is set by JEDI but may be out of sync with Autodesk in some cases.'
+                           .' It only controls UX elements.'),
+
```
Comment (JoshOldenburg) [inline app/Nova/User.php:210]: Checks didn't catch it but the indentation is off here. The content of the lines are fine.

**Review by JoshOldenburg — CHANGES_REQUESTED** [review]
Overall it looks good. There's a few places with minor indentation issues that checks didn't catch. The `UserEditForm` needs a little tweaking.

File: resources/js/components/UserEditForm.vue [hunk 5]
```diff
@@ -267,12 +297,21 @@ export default {
         .put(this.dataUrl, this.user)
         .then(response => {
           this.hasError = false;
-          if (this.user.clickup_email !== this.clickUpEmailInDatabase && this.user.clickup_email && this.user.clickup_email.length > 0) {
+          if ((this.user.clickup_email !== this.clickUpEmailInDatabase && this.user.clickup_email && this.user.clickup_email.length > 0) 
+          && (this.user.autodesk_email !== this.autodeskEmailInDatabase && this.user.autodesk_email && this.user.autodesk_email.length > 0)) {
+            this.clickUpEmailInDatabase = this.user.clickup_email;
+            this.autodeskEmailInDatabase = this.user.autodesk_email;
+            this.feedback = 'Saved! Look out for an email from ClickUp and Autodesk in the next few minutes.'
```
Comment (kberzinch) [inline resources/js/components/UserEditForm.vue:304]:
```suggestion
            this.feedback = 'Saved! Look out for emails from ClickUp and Autodesk in the next few minutes.'
```

**Review by kberzinch — CHANGES_REQUESTED** [review]
This is really close, just a few minor things.

File: resources/js/components/UserEditForm.vue (hunk 5 above)
Comment (CollinAvidano) [reply resources/js/components/UserEditForm.vue:304]: Gosh dangit brain. Thank you for catching this

File: resources/js/components/UserEditForm.vue [hunk 6]
```diff
@@ -199,6 +199,28 @@
...
 
+        <div class="form-group row">
+          <label for="user-autodesk" class="col-sm-2 col-form-label">Autodesk</label>
+          <div class="col-sm-10 col-lg-4">
+            <div class="input-group">
+            <template v-if="user.autodesk_email && user.autodesk_email === autodeskEmailInDatabase">
+              <input v-model="user.autodesk_email" type="text" readonly class="form-control" id="user-autodesk">
+              <div class="input-group-append" v-if="user.autodesk_invite_pending === 1">
+                <a href="/autodesk" class="btn btn-secondary">Resend Invitation</a>
+              </div>
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply resources/js/components/UserEditForm.vue:210]: Per discussion in slack we will add a self service endpoint in JEDI.

File: app/Http/Controllers/AutodeskLibraryController.php [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+                    'message' => 'You are not a member of any teams yet. Join a team first, then try again.',
+                ]
+            );
+        }
+
+        if (null === $user->autodesk_email) {
+            return redirect('/profile');
+        }
+
+        return redirect(config('jedi.host').'/self-service/autodesk');
```
[the remark this answers is not in the document]
Comment (kberzinch) [reply app/Http/Controllers/AutodeskLibraryController.php:37]: Per discussion in slack we will add a self service endpoint in JEDI.

**Review by kberzinch — APPROVED** [review]

File: app/Http/Resources/User.php [hunk 8]
```diff
@@ -64,6 +64,8 @@ public function toArray($request): array
             'clickup_email' => $this->clickup_email,
             'clickup_id' => $this->clickup_id,
             'clickup_invite_pending' => $this->clickup_invite_pending,
+            'autodesk_email' => $this->user->autodesk_email,
+            'autodesk_invite_pending' => $this->user->autodesk_invite_pending,
```
Comment (kberzinch) [inline app/Http/Resources/User.php:68]: Take out `user->` here

**Review by kberzinch — CHANGES_REQUESTED** [review]

**Review by kberzinch — APPROVED** [review]
:shipit:

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**